extension/content/contexts/environment.js

```diff
@@ -196,7 +208,55 @@ export function EnvironmentProvider({ children }) {
           });
         }
       });
-    });
+    };
+
+    window.addEventListener("storage", storageListener);
+
+    // Add event listener for network changes
+    const networkListener = ({ status }) => {
+      console.info("Checking connection status...");
+
+      Object.keys(ENVIRONMENTS).forEach((key) => {
+        dispatch({
+          type: ACTION_SET_CONNECTION_STATUS,
+          status: false,
+          key,
+        });
+      });
+
+      if (status === "up") {
+        Object.entries(ENVIRONMENTS).forEach(async ([key, environment]) => {
+          let status = true;
+          try {
+            await fetchWithTimeout(
```

Comment:
It may also be worth checking that the body is at least vaguely what we expect. Maybe check if it is valid JSON?